app/src/main/java/org/simple/clinic/appconfig/AppConfigModule.kt

```diff
@@ -25,8 +25,8 @@ class AppConfigModule {
   fun provideSelectedCountryPreference(
       rxSharedPreferences: RxSharedPreferences,
       moshi: Moshi
-  ): Preference<Optional<Country>> {
-    val countryPreferenceConverter = MoshiObjectPreferenceConverter(moshi, Country::class.java)
+  ): Preference<Optional<Country_Old>> {
+    val countryPreferenceConverter = MoshiObjectPreferenceConverter(moshi, Country_Old::class.java)
```

Comment:
The eventual plan is to remove providers for `Country_Old` once we finish implementing state selection, right?

Reply:
I am not sure if that would be possible since we would need a type to parse the JSON from preferences. So for now, it will stay for a while? Unless there is a different approach I am missing.

Reply:
Moshi allows us to parse JSON dynamically without creating a type. We could directly read the string from preferences and parse this eventually and remove all traces of the old `Country` model eventually.

Reply:
I would personally take this approach since we don't need the older type for anything except a one-time migration.

Reply:
Ah, we would have to directly use `Types.newParameterizedType` right? Let's do that once the entire feature is done then?